pkg/tsdb/cloudwatch/request_parser.go

```diff
@@ -51,32 +53,31 @@ func (e *cloudWatchExecutor) parseQueries(queries []backend.DataQuery, startTime
 	return requestQueries, nil
 }
 
-// migrateLegacyQuery migrates queries that has a `statistics` field to use the `statistic` field instead.
-// This migration is also done in the frontend, so this should only ever be needed for alerting queries
-// In case the query used more than one stat, the first stat in the slice will be used in the statistic field
-// Read more here https://github.com/grafana/grafana/issues/30629
-func migrateLegacyQuery(queries []backend.DataQuery, startTime time.Time, endTime time.Time) ([]*backend.DataQuery, error) {
+// migrateLegacyQuery is also done in the frontend, so this should only ever be needed for alerting queries
+func migrateLegacyQuery(queries []backend.DataQuery, dynamicLabelsEnabled bool, startTime time.Time, endTime time.Time) ([]*backend.DataQuery, error) {
 	migratedQueries := []*backend.DataQuery{}
 	for _, q := range queries {
 		query := q
-		model, err := simplejson.NewJson(query.JSON)
+		queryJson, err := simplejson.NewJson(query.JSON)
 		if err != nil {
 			return nil, err
 		}
 
-		_, err = model.Get("statistic").String()
-		// If there's not a statistic property in the json, we know it's the legacy format and then it has to be migrated
+		migratedWithStatistic, err := migrateStatisticsToStatistic(queryJson)
 		if err != nil {
-			stats, err := model.Get("statistics").StringArray()
-			if err != nil {
-				return nil, fmt.Errorf("query must have either statistic or statistics field")
-			}
-			model.Del("statistics")
-			model.Set("statistic", stats[0])
-			query.JSON, err = model.MarshalJSON()
-			if err != nil {
-				return nil, err
-			}
+			return nil, err
+		}
+
+		migratedQueryJson := migratedWithStatistic
+
+		_, labelExists := migratedQueryJson.CheckGet("label")
+		if !labelExists && dynamicLabelsEnabled && migratedWithStatistic != nil {
+			migrateAliasToDynamicLabel(migratedWithStatistic)
```

Comment:
`migrateStatisticsToStatistic` is now consistent with `migrateAliasToDynamicLabel`
Since they're passing around a pointer, they're changing the query in place and don't need the return value